### PR TITLE
Add docstring

### PR DESCRIPTION
**Pull request by sbrunner**

### CODE REVIEW

File: c2cciutils/__init__.py [hunk 1]
```diff
@@ -434,6 +482,15 @@ def print_versions(config: c2cciutils.configuration.PrintVersions) -> bool:
 
 
 def gopass(key: str, default: Optional[str] = None) -> Optional[str]:
+    """
+    Get a value in gopass.
```
Comment (arnaud-morvan) [inline c2cciutils/__init__.py:486]:
```suggestion
    Get a value from gopass.
```

File: c2cciutils/__init__.py [hunk 2]
```diff
@@ -404,19 +442,29 @@ def match(
 
 def get_value(matched: Optional[Match[str]], config: Optional[VersionTransform], value: str) -> str:
     """
-    Get the final value
+    Get the final value.
 
     `match`, `config` and `value` are the result of `match`.
 
     The `config` should have a `to` ad a expand template.
```
Comment (arnaud-morvan) [inline c2cciutils/__init__.py:449]:
```suggestion
    The `config` should have a `to` key with an expand template.
```

File: c2cciutils/__init__.py [hunk 3]
```diff
@@ -443,10 +500,25 @@ def gopass(key: str, default: Optional[str] = None) -> Optional[str]:
...
+    Arguments:
+        secret: The secret value
+        key: The key
+    """
     subprocess.check_output(["gopass", "insert", "--force", key], input=secret.encode())
 
 
 def add_authorization_header(headers: Dict[str, str]) -> Dict[str, str]:
+    """
+    Add the Authorization needec to be consider as connected on GitHub.
```
Comment (arnaud-morvan) [inline c2cciutils/__init__.py:515]:
```suggestion
    Add the Authorization header needed to be authenticated on GitHub.
```

File: c2cciutils/__init__.py [hunk 4]
```diff
@@ -460,10 +532,12 @@ def add_authorization_header(headers: Dict[str, str]) -> Dict[str, str]:
 
 def graphql(query_file: str, variables: Dict[str, Any], default: Any = None) -> Any:
     """
-    Get the result a a graphql on GitHub
+    Get the result a a graphql on GitHub.
```
Comment (arnaud-morvan) [inline c2cciutils/__init__.py:535]:
```suggestion
    Get a graphql result from GitHub.
```

File: c2cciutils/__init__.py [hunk 5]
```diff
@@ -460,10 +532,12 @@ def add_authorization_header(headers: Dict[str, str]) -> Dict[str, str]:
...
 
-    query_file: the file related this module contains the GraphQL querry
-    variables: the query variables
+    Arguments:
+        query_file: The file related this module contains the GraphQL querry
```
Comment (arnaud-morvan) [inline c2cciutils/__init__.py:538]:
```suggestion
        query_file: Relative path from this file to the GraphQL query file.
```

File: c2cciutils/__init__.py [hunk 6]
```diff
@@ -460,10 +532,12 @@ def add_authorization_header(headers: Dict[str, str]) -> Dict[str, str]:
...
+        variables: The query variables
+        default:  The return result if we are not autorised to get the resource
```
Comment (arnaud-morvan) [inline c2cciutils/__init__.py:540]:
```suggestion
        default:  The return result if we are not authorized to get the resource
```

File: c2cciutils/__init__.py [hunk 7]
```diff
@@ -502,9 +576,11 @@ def get_git_files_mime(
     mime_type: str = "text/x-python", ignore_patterns_re: Optional[List[str]] = None
 ) -> List[str]:
     """
-    Get all the files in git that have the specified mime type
+    Get all the files in git that have the specified mime type.
```
Comment (arnaud-morvan) [inline c2cciutils/__init__.py:579]:
```suggestion
    Get list of paths from git with all the files that have the specified mime type.
```

File: c2cciutils/__init__.py [hunk 8]
```diff
@@ -502,9 +576,11 @@ def get_git_files_mime(
...
 
-    ignore_patterns_re: list of regular expression to be ignored
+    Arguments:
+        mime_type: The used MIME type
```
Comment (arnaud-morvan) [inline c2cciutils/__init__.py:582]:
```suggestion
        mime_type: The considered MIME type
```

File: c2cciutils/__init__.py [hunk 9]
```diff
@@ -502,9 +576,11 @@ def get_git_files_mime(
...
+        ignore_patterns_re: A list of regular expression of files that we should ignore
```
Comment (arnaud-morvan) [inline c2cciutils/__init__.py:583]:
```suggestion
        ignore_patterns_re: A list of regular expressions of files that we should ignore
```

File: c2cciutils/audit.py [hunk 10]
```diff
@@ -25,7 +28,12 @@ def print_versions(
     args: Namespace,
 ) -> bool:
     """
-    Print the versions
+    Print the versions.
+
+    Arguments:
+        config: The config
```
Comment (arnaud-morvan) [inline c2cciutils/audit.py:34]: Not sure about the difference between "The config" and "The full config".

File: c2cciutils/audit.py [hunk 11]
```diff
@@ -224,6 +252,9 @@ def pipenv(
 
 
 def get_global_npm_cve() -> List[int]:
+    """
+    Get the CVE that installed gobally by GitHub.
```
Comment (arnaud-morvan) [inline c2cciutils/audit.py:256]:
```suggestion
    Get the CVE which were installed globally by GitHub.
```

File: c2cciutils/checks.py [hunk 12]
```diff
@@ -126,11 +149,16 @@ def editorconfig(
     args: Namespace,
 ) -> bool:
     """
-    Check the right editorconfig configuration
+    Check the right editorconfig configuration.
```
Comment (arnaud-morvan) [inline c2cciutils/checks.py:152]:
```suggestion
    Check the editorconfig configuration.
```
For sure we check it's right

File: c2cciutils/checks.py [hunk 13]
```diff
@@ -197,7 +230,12 @@ def gitattribute(
 
 def eof(config: None, full_config: c2cciutils.configuration.Configuration, args: Namespace) -> bool:
     """
-    Check the files eof
+    Check the files end of file.
```
Comment (arnaud-morvan) [inline c2cciutils/checks.py:233]:
```suggestion
    Check the non empty text files end with "\n".
```

File: c2cciutils/checks.py [hunk 14]
```diff
@@ -247,11 +285,16 @@ def workflows(
     args: Namespace,
 ) -> bool:
     """
-    Do some generic check on the workflows
+    Do some generic check on the workflows.
```
Comment (arnaud-morvan) [inline c2cciutils/checks.py:288]:
```suggestion
    Check each workflow have a timeout and/or do not use blacklisted images.
```
Sorry but I do not know what is a generic check on a workflow.

File: c2cciutils/checks.py [hunk 15]
```diff
@@ -290,7 +333,7 @@ def required_workflows(
     args: Namespace,
 ) -> bool:
     """
-    Test that we have the required workflow with the required element
+    Test that we have the required workflow with the required element.
```
Comment (arnaud-morvan) [inline c2cciutils/checks.py:336]:
```suggestion
    Test we have the required workflow with the required properties.
```

File: c2cciutils/checks.py [hunk 16]
```diff
@@ -413,8 +461,9 @@ def versions(
     args: Namespace,
 ) -> bool:
     """
-    Verify that various GitHub / CI tools versions or branches configuration match with versions
-    from `SECURITY.md` file.
+    Verify that various GitHub / CI tools versions or branches configuration.
```
Comment (arnaud-morvan) [inline c2cciutils/checks.py:464]: Missing verb for complete sentence: Verify that ...
```suggestion
    Verify various GitHub / CI tools versions or branches configurations.
```

File: c2cciutils/checks.py [hunk 17]
```diff
@@ -413,8 +461,9 @@ def versions(
...
+
+    That match with versions from `SECURITY.md` file.
```
Comment (arnaud-morvan) [inline c2cciutils/checks.py:466]: What is that ?
```suggestion
    Versions from ... match with versions from `SECURITY.md` file.
```

File: c2cciutils/checks.py [hunk 18]
```diff
@@ -531,7 +593,12 @@ def _versions_rebuild(
     full_config: c2cciutils.configuration.Configuration,
 ) -> bool:
     """
-    Check that the rebuild branches correspond to the version from the Security.md
+    Check that the rebuild branches correspond to the version from the Security.md.
```
Comment (arnaud-morvan) [inline c2cciutils/checks.py:596]:
```suggestion
    Check the rebuild branches match with the versions from the Security.md.
```

File: c2cciutils/checks.py [hunk 19]
```diff
@@ -567,7 +634,11 @@ def _versions_backport_labels(
     all_versions: Set[str], full_config: c2cciutils.configuration.Configuration
 ) -> bool:
     """
-    Check that the backport labels correspond to the version from the Security.md
+    Check that the backport labels correspond to the version from the Security.md.
```
Comment (arnaud-morvan) [inline c2cciutils/checks.py:637]:
```suggestion
    Check the backport labels match with the version from the Security.md.
```

File: c2cciutils/checks.py [hunk 20]
```diff
@@ -490,7 +548,11 @@ def _get_branch_matrix(
 
 def _versions_audit(all_versions: Set[str], full_config: c2cciutils.configuration.Configuration) -> bool:
     """
-    Check that the audit branches correspond to the version from the Security.md
+    Check that the audit branches correspond to the version from the Security.md.
```
Comment (arnaud-morvan) [inline c2cciutils/checks.py:551]:
```suggestion
    Check the audit branches match with the versions from the Security.md.
```

File: c2cciutils/checks.py [hunk 21]
```diff
@@ -607,7 +678,11 @@ def _versions_backport_labels(
 
 def _versions_branches(all_versions: Set[str], full_config: c2cciutils.configuration.Configuration) -> bool:
     """
-    Check that the branches correspond to the version from the Security.md
+    Check that the branches correspond to the version from the Security.md.
```
Comment (arnaud-morvan) [inline c2cciutils/checks.py:681]:
```suggestion
    Check the branches match with the versions from the Security.md.
```

File: c2cciutils/checks.py [hunk 22]
```diff
@@ -742,13 +827,18 @@ def codespell(
     args: Namespace,
 ) -> bool:
     """
-    Run codespell check on all files
+    Run codespell check on all files.
 
     If therer is an `spell-ignore-words.txt` file we consider it with ignore word
```
Comment (arnaud-morvan) [inline c2cciutils/checks.py:832]:
```suggestion
    If there is an `spell-ignore-words.txt` file we consider it with ignore word
```

File: c2cciutils/checks.py [hunk 23]
```diff
@@ -1035,7 +1137,12 @@ def print_versions(
     args: Namespace,
 ) -> bool:
     """
-    Print some tools version
+    Print some tools version.
```
Comment (arnaud-morvan) [inline c2cciutils/checks.py:1140]:
```suggestion
    Print some tools versions.
```

File: c2cciutils/publish.py [hunk 24]
```diff
@@ -19,8 +23,15 @@
 
 
 class GoogleCalendar:
+    """
+    Interact with the Goocle Calendar API.
```
Comment (arnaud-morvan) [inline c2cciutils/publish.py:27]:
```suggestion
    Interact with the Google Calendar API.
```

File: c2cciutils/publish.py [hunk 25]
```diff
@@ -130,6 +156,13 @@ def create_event(
         summary: str = "dummy/image:{}".format(datetime.datetime.now().isoformat()),
         description: str = "description",
     ) -> None:
+        """
+        Create a calendart event.
```
Comment (arnaud-morvan) [inline c2cciutils/publish.py:160]:
```suggestion
        Create a calendar event.
```

File: c2cciutils/publish.py [hunk 26]
```diff
@@ -144,10 +177,16 @@ def create_event(
         print("created event with id: {}".format(event_result["id"]))
 
     def _print_credentials(self) -> None:
+        """
+        Print the credentails.
```
Comment (arnaud-morvan) [inline c2cciutils/publish.py:181]:
```suggestion
        Print the credentials.
```

File: c2cciutils/publish.py [hunk 27]
```diff
@@ -144,10 +177,16 @@ def create_event(
...
+        """
         # UNSAFE: DO NEVER PRINT CREDENTIALS IN CI ENVIRONMENT, DEBUG ONLY!!!!
         print(self.creds.to_json())
 
     def save_credentials_to_gopass(self) -> None:
+        """
+        Sabe the calendar credentials to gopass.
```
Comment (arnaud-morvan) [inline c2cciutils/publish.py:188]:
```suggestion
        Save the calendar credentials to gopass.
```

File: c2cciutils/publish.py [hunk 28]
```diff
@@ -161,11 +200,17 @@ def save_credentials_to_gopass(self) -> None:
             c2cciutils.gopass_put(secret, key)
 
     def __del__(self) -> None:
+        """
+        Delete.
```
Comment (arnaud-morvan) [inline c2cciutils/publish.py:204]: Delete what ?

File: c2cciutils/scripts/audit.py [hunk 29]
```diff
@@ -1,13 +1,20 @@
...
+
 import argparse
 import sys
 
 import c2cciutils.audit
 
 
 def main() -> None:
+    """
+    Run the main function.
```
Comment (arnaud-morvan) [inline c2cciutils/scripts/audit.py:16]: Run the audit ?

File: c2cciutils/scripts/checks.py [hunk 30]
```diff
@@ -1,13 +1,20 @@
...
+
 import argparse
 import sys
 
 import c2cciutils.checks
 
 
 def main() -> None:
+    """
+    Run the main function.
```
Comment (arnaud-morvan) [inline c2cciutils/scripts/checks.py:16]: Run the checks.

File: c2cciutils/scripts/clean.py [hunk 31]
```diff
@@ -14,7 +17,12 @@
 
 def clean(image: str, tag: str, token: str) -> None:
     """
-    Delete an image on Docker hub
+    Delete an image on Docker hub.
```
Comment (arnaud-morvan) [inline c2cciutils/scripts/clean.py:20]:
```suggestion
    Delete an image from Docker hub.
```

File: c2cciutils/scripts/clean.py [hunk 32]
```diff
@@ -41,6 +49,9 @@ def clean(image: str, tag: str, token: str) -> None:
 
 
 def main() -> None:
+    """
+    Run the main function.
```
Comment (arnaud-morvan) [inline c2cciutils/scripts/clean.py:53]: For sure when I call `main()` it run the main function.

File: c2cciutils/scripts/main.py [hunk 33]
```diff
@@ -10,6 +14,9 @@
 
 
 def main() -> None:
+    """
+    Run the main function.
```
Comment (arnaud-morvan) [inline c2cciutils/scripts/main.py:18]: Hoow, surprise, when I call `main()` it run the main function ;-)

File: c2cciutils/scripts/publish.py [hunk 34]
```diff
@@ -39,6 +52,9 @@ def to_version(full_config: c2cciutils.configuration.Configuration, value: str,
 
 
 def main() -> None:
+    """
+    Run the main function.
```
Comment (arnaud-morvan) [inline c2cciutils/scripts/publish.py:56]: Another "Run the main function".
Note you can copy/paste the description on following line.

File: c2cciutils/security.py [hunk 35]
```diff
@@ -1,5 +1,8 @@
 # -*- coding: utf-8 -*-
 
+"""
+Read the table of versions got from SECURITY.md.
```
Comment (arnaud-morvan) [inline c2cciutils/security.py:4]:
```suggestion
Read the table of versions from SECURITY.md.
```

File: c2cciutils/security.py [hunk 36]
```diff
@@ -10,7 +13,7 @@
 
 class Security:
     """
-    Read the table of versions got from SECURITY.md
+    Read the table of versions got from SECURITY.md.
```
Comment (arnaud-morvan) [inline c2cciutils/security.py:16]:
```suggestion
    Read the table of versions from SECURITY.md.
```

File: c2cciutils/security.py [hunk 37]
```diff
@@ -19,7 +22,9 @@ class Security:
 
     def __init__(self, status: str):
         """
-        Parameters:
+        Initialisatise.
```
Comment (arnaud-morvan) [inline c2cciutils/security.py:25]:
```suggestion
        Initialize.
```

File: c2cciutils/security.py [hunk 38]
```diff
@@ -19,7 +22,9 @@ class Security:
...
+
+        Arbuments:
```
Comment (arnaud-morvan) [inline c2cciutils/security.py:27]:
```suggestion
        Arguments:
```

File: setup.py [hunk 39]
```diff
@@ -14,6 +18,9 @@
 
 
 def long_description() -> str:
+    """
+    Get the long description.
```
Comment (arnaud-morvan) [inline setup.py:22]:
```suggestion
    Get the long description from README.md file.
```

**Review by arnaud-morvan — APPROVED** [review]
Huge improvement, thanks, found some typos.

File: c2cciutils/audit.py (hunk 10 above)
Comment (sbrunner) [reply c2cciutils/audit.py:34]: => 
```
        config: The audit section config
        full_config: All the CI config
```

File: c2cciutils/checks.py (hunk 17 above)
Comment (sbrunner) [reply c2cciutils/checks.py:466]: =>
    Versions from audit workflow, rebuild workdlow(s), protected branches and backport lables 
    match with versions from `SECURITY.md` file.

File: c2cciutils/publish.py (hunk 28 above)
Comment (sbrunner) [reply c2cciutils/publish.py:204]: The new check requires a docstring...

Comment (sbrunner) [reply c2cciutils/publish.py:204]: Or I remove all of them:
D105 | Missing docstring in magic method

**Pull request merged**